azurelinuxagent/common/utils/restutil.py

```diff
@@ -513,7 +513,7 @@ def http_request(method,
                 raise
             IOErrorCounter.increment(host=host, port=port)
             clean_url = _trim_url_parameters(url)
-            msg = '[HTTP Failed] {0} {1} -- IOError {2}'.format(method, clean_url, e)
+            msg = '[HTTP Failed] {0} {1} -- IOError {2}'.format(method, redact_sas_tokens_in_urls(clean_url), e)
```

Comment:
this should be done in event.py instead, right?

Reply:
we shouldn't write to agent logs as well

Reply:
Updated: Now, every event and log will go through sanitization